src/sv-pipeline/scripts/downstream_analysis_and_filtering/clean_frequencies_table.R

```diff
@@ -11,56 +11,52 @@
-clean.ACs <- function(ACs){
-  ACs.to.clean <- grep(",",ACs,fixed=T)
-  cleaned.ACs <- as.vector(as.numeric(sapply(ACs[ACs.to.clean],function(s){
-    return(sum(as.numeric(unlist(strsplit(s,split=",")))[-2]))
-  })))
-  ACs[ACs.to.clean] <- cleaned.ACs
-  return(as.numeric(ACs))
-}
 #Import a single table of freq data
-import.freq.table <- function(path,pops){
+import.freq.table <- function(path, pops){
   #Read data
   dat <- read.table(path,header=T,comment.char="",sep="\t")
   colnames(dat)[1:3] <- c("VID","SVLEN","SVTYPE")
-  #Clean multiallelic ACs
-  dat[,grep("_AC",colnames(dat),fixed=T)] <- apply(dat[,grep("_AC",colnames(dat),fixed=T)],2,clean.ACs)
-  #Convert all ANs to numerics
-  dat[,grep("_AN",colnames(dat),fixed=T)] <- apply(dat[,grep("_AN",colnames(dat),fixed=T)],2,as.numeric)
-  #Double counts for MCNVs (since first allele for all MCNVs is nulled out)
-  MCNV.idx <- grep("_MCNV_",dat[,1],fixed=T)
-  dat[MCNV.idx,grep("_AC",colnames(dat),fixed=T)] <- 2*dat[MCNV.idx,grep("_AC",colnames(dat),fixed=T)]
-  dat[MCNV.idx,grep("_AN",colnames(dat),fixed=T)] <- 2*dat[MCNV.idx,grep("_AN",colnames(dat),fixed=T)]
+  #Convert multiallelic CN_NONREF_COUNT and CN_NUMBER to AC and AN, and drop CN_ columns
+  mcnv.idxs <- which(dat$SVTYPE %in% c("CNV", "MCNV"))
+  dat[mcnv.idxs, grep("_AC",colnames(dat),fixed=T)] <- dat[mcnv.idxs, grep("_CN_NONREF_COUNT",colnames(dat),fixed=T)]
+  dat[mcnv.idxs, grep("_AN",colnames(dat),fixed=T)] <- dat[mcnv.idxs, grep("_CN_NUMBER",colnames(dat),fixed=T)]
+  dat <- dat[, -union(grep("_CN_NONREF_COUNT", colnames(dat), fixed=T), grep("_CN_NUMBER", colnames(dat), fixed=T))]
+  #Convert all ANs and ACs to numerics
+  dat[, -c(1:3)] <- apply(dat[, -c(1:3)], 2, as.numeric)
   #Adjust calls on sex chromosomes
   for(pop in pops){
     x.idx <- unique(c(grep("_X_",dat[,1],fixed=T),
                       grep("_chrX_",dat[,1],fixed=T)))
     y.idx <- unique(c(grep("_Y_",dat[,1],fixed=T),
                       grep("_chrY_",dat[,1],fixed=T)))
-    master.AC.idx <- which(colnames(dat)==paste(pop,"AC",sep="_"))
-    master.AN.idx <- which(colnames(dat)==paste(pop,"AN",sep="_"))
-    male.AC.idx <- which(colnames(dat)==paste(pop,"MALE","AC",sep="_"))
-    male.AN.idx <- which(colnames(dat)==paste(pop,"MALE","AN",sep="_"))
-    female.AC.idx <- which(colnames(dat)==paste(pop,"FEMALE","AC",sep="_"))
-    female.AN.idx <- which(colnames(dat)==paste(pop,"FEMALE","AN",sep="_"))
-    #For variants on chrX, overwrite master counts with female-specific counts
-    dat[x.idx,master.AC.idx] <- dat[x.idx,female.AC.idx]
-    dat[x.idx,master.AN.idx] <- dat[x.idx,female.AN.idx]
-    #For all variants on chrY, overwrite master counts with male-specific counts
-    dat[y.idx,master.AC.idx] <- dat[y.idx,male.AC.idx]
-    dat[y.idx,master.AN.idx] <- dat[y.idx,male.AN.idx]
+    if(length(c(x.idx, y.idx)) > 0){
+      if(4 == length(intersect(colnames(dat),
+                               c(paste(pop,"MALE","AC",sep="_"),
```

Comment:
A tiny nitpick, but It'd be a little cleaner to construct these names once, before the if statement, rather than constructing them all twice.

Reply:
No problem — done.